endpoints/openrtb2/auction.go

```diff
@@ -1516,6 +1522,25 @@ func setAuctionTypeImplicitly(r *openrtb_ext.RequestWrapper) {
 	}
 }
 
+func setGPCImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) error {
+	secGPC := httpReq.Header.Get(secGPCHdrKey)
+	fmt.Printf("Sec-GPC Header: %s\n", secGPC)
+
+	if secGPC != "1" {
+		return nil
+	}
+
+	regExt, err := r.GetRegExt()
+	if err != nil {
+		return err
+	}
+
+	gpc := "1"
+	regExt.SetGPC(&gpc)
```

Comment:
I suggest adding the following early return just before the set:
```
if regExt.GetGPC() != nil {
    return nil
}
```
This way we won't perform an unnecessary write which sets the dirty flag and causes extra work when rebuilding the request downstream.

Reply:
Hi @przemkaczmarek, here is a super nitpick to this function. 
In the issue description the logic is straightforward:
```
1. If the incoming request already contains regs.ext.gpc, use that.
2. Otherwise, if there's an HTTP header for Sec-GPC with a value of "1" or 1, set regs.ext.gpc to "1".
```
 The code here does the right thing, however it's a little confusing and difficult to read. I had to read it couple of times to understand it.
Here is a suggestion that I think traces better to the requirements and might be better for readability:
```go
func setGPCImplicitly(httpReq *http.Request, r *openrtb_ext.RequestWrapper) error {

	regExt, err := r.GetRegExt()
	if err != nil {
		return err
	}

	if regExt.GetGPC() != nil {
		return nil
	}

	secGPC := httpReq.Header.Get(secGPCKey)

	if secGPC == "1" {
		gpc := "1"
		regExt.SetGPC(&gpc)
	}

	return nil
}

```
The unit test passes as is. 

What do you think?

Reply:
In my version of the code, I first check the value of secGPC, and if it doesn't meet the condition, the function ends immediately with return nil. In your version of the code, this happens later, which leads to unnecessary retrieval of regExt, even when it's not needed. I avoid unnecessary operations. I avoid performing costly operations (such as r.GetRegExt()) if the secGPC header does not have the value '1'. In the first version, you retrieve the regExt extension regardless of the secGPC value, which is less optimal

Reply:
Yes, I understand this. I am only concerned about the readability. 
Also this is a minor performance optimization (but still an optimization!) so it's up to you to change it. 
I approve this PR, and if you decide to change it - I'll re-approve it right away too.

Reply:
I prefer to keep the code as is but thanks for the comment